src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -2787,20 +2805,58 @@ int wm_vuldet_insert(wm_vuldet_db *parsed_oval, update_node *update) {
             sqlite3_bind_int(stmt, 7, 0);
             sqlite3_bind_int(stmt, 8, vul_it->ignore);
             sqlite3_bind_int(stmt, 9, 0);
+            if (vul_it->deps) {
+                // Check the dependency array to avoid inserting more IDs than necessary
+                if (deps_size < vul_it->deps->elements) {
+                    deps_size = vul_it->deps->elements;
+                }
+                for (int i = 0; i < deps_size; ++i) {
+                    if (!dependency || dependency[i] != vul_it->deps->test_ref[i]) {
+                        os_free(dependency);
+                        deps_size = vul_it->deps->elements;
+                        dependency = vul_it->deps->test_ref;
```

Comment:
Possible memory leak: This memory doesn't seem to be freed in the last iteration of the loop.